openedx/core/djangoapps/content_tagging/rest_api/v1/objecttag_export_helpers.py

```diff
@@ -38,51 +42,136 @@ def iterate_with_level(
+        display_name=course.display_name_with_default,
         block_id=course_id,
         category=course.category,
-        object_tags=object_tag_cache.get(str(content_key), {}),
+        object_tags=object_tag_cache.get(course_id, {}),
+        children=None,
+    )
+
+    return tagged_course, course.children if course.has_children else []
+
+
+def get_library_tagged_object_and_children(
+    library_key: LibraryLocatorV2, object_tag_cache: ObjectTagByObjectIdDict
+) -> tuple[TaggedContent, list[LibraryXBlockMetadata]]:
+    """
+    Returns a TaggedContent with library metadata with its tags, and its children.
+    """
+    library = library_api.get_library(library_key)
+    assert library is not None
+
+    library_id = str(library_key)
+
+    tagged_library = TaggedContent(
+        display_name=library.title,
+        block_id=library_id,
+        category='library',
+        object_tags=object_tag_cache.get(library_id, {}),
+        children=None,
+    )
+
+    children = library_api.get_library_blocks(library_key)
+
+    return tagged_library, children
+
+
+def get_xblock_tagged_object_and_children(
+    usage_key: UsageKey, object_tag_cache: ObjectTagByObjectIdDict, store
+) -> tuple[TaggedContent, list[XBlock]]:
+    """
+    Returns a TaggedContent with xblock metadata with its tags, and its children.
+    """
+    block = store.get_item(usage_key)
+    block_id = str(usage_key)
+    tagged_block = TaggedContent(
+        display_name=block.display_name_with_default,
+        block_id=block_id,
+        category=block.category,
+        object_tags=object_tag_cache.get(block_id, {}),
         children=None,
     )
 
-    blocks = [(tagged_course, course)]
+    return tagged_block, block.children if block.has_children else []
+
+
+def get_library_block_tagged_object(
+    library_block: LibraryXBlockMetadata, object_tag_cache: ObjectTagByObjectIdDict
+) -> tuple[TaggedContent, None]:
+    """
+    Returns a TaggedContent with library content block metadata and its tags,
+    and 'None' as children.
+    """
+    block_id = str(library_block.usage_key)
+    tagged_library_block = TaggedContent(
+        display_name=library_block.display_name,
+        block_id=block_id,
+        category=library_block.usage_key.block_type,
+        object_tags=object_tag_cache.get(block_id, {}),
+        children=None,
+    )
+
+    return tagged_library_block, None
+
+
+def build_object_tree_with_objecttags(
+    content_key: LibraryLocatorV2 | CourseKey,
+    object_tag_cache: ObjectTagByObjectIdDict,
+) -> TaggedContent:
+    """
+    Returns the object with the tags associated with it.
+    """
+    if isinstance(content_key, CourseKey):
+        tagged_content, children = get_course_tagged_object_and_children(
+            content_key, object_tag_cache
+        )
+    elif isinstance(content_key, LibraryLocatorV2):
+        tagged_content, children = get_library_tagged_object_and_children(
+            content_key, object_tag_cache
+        )
```

Comment:
Up here we can decide which method to use when fetching child blocks, so we don't have to do the type checks in the loop below:

```suggestion
    if isinstance(content_key, CourseKey):
        tagged_content, children = get_course_tagged_object_and_children(
            content_key, object_tag_cache
        )
        get_tagged_children = get_xblock_tagged_object_and_children
    elif isinstance(content_key, LibraryLocatorV2):
        tagged_content, children = get_library_tagged_object_and_children(
            content_key, object_tag_cache
        )
        get_tagged_children = get_library_block_tagged_object
```

But you'll need to remove the `store` parameter from `get_xblock_tagged_object_and_children` and just call this again (it's ok to do this, the modulestore is globally cached):

```
    store = modulestore()
```